bindings/java/src/org/sleuthkit/datamodel/SleuthkitJNI.java

```diff
@@ -674,7 +704,13 @@ public static long openFile(long fsHandle, long fileId, TSK_FS_ATTR_TYPE_ENUM at
 		 * need to convert negative attribute id to uint16 which is what TSK is
 		 * using to store attribute id.
 		 */
-		long fileHandle = openFileNat(fsHandle, fileId, attrType.getValue(), convertSignedToUnsigned(attrId));
+		long fileHandle = 0;
+		getReadLock();
+		try {
+			fileHandle = openFileNat(fsHandle, fileId, attrType.getValue(), convertSignedToUnsigned(attrId));			
+		} finally {
+			releaseReadLock();
+		}
 		HandleCache.addFileHandle(fileHandle, fsHandle);
```

Comment:
Should this be in the lock?  Similar code is in other places.

Reply:
This is because addFileHandle currently gets the write lock and we can't hold both read and write at the same time. If we can get rid of most of the write locks this will no longer have to be outside the lock.